code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplate.kt

```diff
@@ -41,276 +39,184 @@ internal sealed class AEPPushTemplate {
-            count?.let {
-                badgeCount = count.toInt()
-            }
-        } catch (e: NumberFormatException) {
-            Log.debug(
-                PushTemplateConstants.LOG_TAG,
-                SELF_TAG,
-                "Exception in converting notification badge count to int - ${e.localizedMessage}."
-            )
-        }
-        notificationPriority = NotificationPriority.getNotificationCompatPriorityFromString(
-            DataReader.optString(
-                data,
-                PushTemplateConstants.PushPayloadKeys.NOTIFICATION_PRIORITY,
-                null
-            )
-        )
-        notificationImportance =
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) getNotificationImportanceFromString(
-                DataReader.optString(
-                    data,
-                    PushTemplateConstants.PushPayloadKeys.NOTIFICATION_PRIORITY,
-                    null
-                )
-            ) else null
-
-        notificationVisibility = NotificationVisibility.getNotificationCompatVisibilityFromString(
-            DataReader.optString(
-                data,
-                PushTemplateConstants.PushPayloadKeys.NOTIFICATION_VISIBILITY,
-                null
-            )
-        )
-        channelId =
-            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.CHANNEL_ID, null)
-        templateType =
-            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TEMPLATE_TYPE, null)
-                ?.let { PushTemplateType.fromString(it) }
-        tag = DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TAG, null)
-        val stickyValue =
-            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.STICKY, null)
-        isNotificationSticky =
-            if (stickyValue.isNullOrEmpty()) false else java.lang.Boolean.parseBoolean(
-                stickyValue
-            )
-        ticker = DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TICKER, null)
-        expandedBodyText = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.EXPANDED_BODY_TEXT, null
-        )
-        expandedBodyTextColor = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.EXPANDED_BODY_TEXT_COLOR, null
-        )
-        titleTextColor = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.TITLE_TEXT_COLOR, null
-        )
-        smallIconColor = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.SMALL_ICON_COLOR, null
-        )
-        notificationBackgroundColor = DataReader.optString(
-            data,
-            PushTemplateConstants.PushPayloadKeys.NOTIFICATION_BACKGROUND_COLOR,
-            null
-        )
-        isFromIntent = false
+    init {
+        // extract the notification payload version
+        payloadVersion = data.getString(PushPayloadKeys.VERSION)?.toInt()
```

Comment:
Can we standardize the type for `payloadVersion`. If we need an `Int` then lets use intent.getIntExtra(...) internally and add it as an int to the bundle as well.

Reply:
Here is the challenge, Since the remoteMessage only give us the Map<String,String> where all the notification payload values are in string.

I need to maintain the same type for Intent as well because now we are having a common constructor

Reply:
Ah! I see. Got it. 👍🏼

Reply:
I see. Does it make sense to change the type of  `payloadVersion`  to String in `AEPPushTemplate` then?
Or does the schema from ACC define it explicitly to be a string?

Reply:
Verifying the specs in wiki. Looks like payload version seems to a string value. Will change the datatype to keep them as String always